sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1618,6 +1617,7 @@ static void *_gnrc_netif_thread(void *args)
         dev->context = NULL;
         return NULL;
     }
+    netif_register(&netif->netif);
```

Comment:
Above there is some handling labeled as `/* unregister this netif instance */`, seems like the comment should be fixed

Reply:
changed it to *invalidate* - I wonder if this is even necessary anymore if we simple don't register the interface to begin with.

Reply:
I wonder the same, but don't dare to affirm it hahaha, maybe @miri64 is bolder than me :)

Reply:
>  I wonder if this is even necessary anymore if we simple don't register the interface to begin with.

Does this mean not calling `netif_register()`?

Reply:
Yes, `netif_register()` is only called if `init()` did succeed with this PR.

Reply:
yes, I know, but I was asking wether that means avoid calling `netif_register` at all or not.
I think we can easily get out of this pattern of unregistering the netif if the device is initialized first and then GNRC fills in the callbacks and whatever is required there. I agree that initializing the interface before knowing if the device is ready, as it is, is probably not the best solution

Reply:
> I wonder the same, but don't dare to affirm it hahaha, maybe @miri64 is bolder than me :)

Nope, that is @jia200x's code, so he has to be bold here ;-P

Reply:
TBH I didn't understand the original question (as stated above), but I can try to explain why we added this `netif_register`.

- In the beginning we set the number of interfaces with the GNRC_NETIF_NUMOF macro. This was really hard to maintain and we always ended up doing some build system tricks in order to add interfaces (e.g via `netdev_default`). We agreed then in using a linked list to add interfaces on the fly (and thus we added this `netif_register`).
- We decided to do at `netif` level (and not `gnrc_netif`) because this logic would be used to iterate OS interfaces, regardless of the network stack (LWIP, GNRC, OpenThread). Also, we thought that we could reuse code if all network-stack-specific descriptors inherit from `netif_t`.

There are different ways to implement this pattern, so if someone has a better idea please speak up.
Also, as described above, I strongly believe we should first init the device and _then_ try to setup the interface. Besides the fact this goes more in the direction of "setting the interface up", it saves as from these de-initialization lines.

Reply:
The question was: if `netif` is never used again (because it was not added to the list if init failed), why do we need to set it's members to zero?

Reply:
ah... yeah, I don't think we need to do that. I haven't digged deeper, but I don't see why this could cause an issue.

Reply:
hmmmm from what I see this was probably required when we were still using GNRC_NETIF_NUMOF. Now it shouldn't be required

Reply:
Ok, I removed that part 1c2f736